File: LeetCode/1232. Check If It Is a Straight Line/solution.cpp
/**
 * 79 / 79 test cases passed.
 * Status: Accepted
 * Runtime: 8 ms
 */
class Solution {
public:
    bool checkStraightLine(vector<vector<int>>& coordinates) {
        int deltaX = coordinates[0][0], deltaY = coordinates[0][1];
        for (int i = 0; i < coordinates.size(); ++ i) {
            coordinates[i][0] -= deltaX;
            coordinates[i][1] -= deltaY;
        }
        // int k = (int) coordinates[1][1] / coordinates[1][0];
        // (x-x1)/(x2-x1)=(y-y1)/(y2-y1), x1=0,y1=0
        // y2 x = x1 y => Ax = By
        int A = coordinates[1][1], B = coordinates[1][0];
        for (int i = 2; i < coordinates.size(); ++ i) {
            if (A * coordinates[i][0] != B * coordinates[i][1]) {
                return false;
            }
        }
        return true;
    }
};

File: LeetCode/1232. Check If It Is a Straight Line/Solution.java
/**
 * 79 / 79 test cases passed.
 * Status: Accepted
 * Runtime: 0 ms
 */
class Solution {
    public boolean checkStraightLine(int[][] coordinates) {
        if (coordinates.length <= 2) return true;
        int x = coordinates[0][0], y = coordinates[0][1];
        int dx = coordinates[1][0] - x;
        int dy = coordinates[1][1] - y;
        for (int i = 2; i < coordinates.length; ++ i) {
            int _dx = coordinates[i][0] - x;
            int _dy = coordinates[i][1] - y;
            if (dx * _dy != _dx * dy) return false;
        }
        return true;
    }
}